azurelinuxagent/common/rdma.py

```diff
@@ -90,3 +96,133 @@ def reboot_system(self):
            the rdma driver cannot be unloaded with rmmod"""
         logger.info('System reboot')
         shellutil.run('shutdown -r now')
+
+dapl_config_paths = [
+    '/etc/dat.conf', '/etc/rdma/dat.conf', '/usr/local/etc/dat.conf']
+
+class RDMADeviceHandler(object):
+
+    """
+    Responsible for writing RDMA IP and MAC address to the /dev/hvnd_rdma
+    interface.
+    """
+
+    rdma_dev = '/dev/hvnd_rdma'
+    device_check_timeout_sec = 120
+    device_check_interval_sec = 1
+
+    ipv4_addr = None
+    mac_adr = None
+
+    def __init__(self, ipv4_addr, mac_addr):
+        self.ipv4_addr = ipv4_addr
+        self.mac_addr = mac_addr
+
+    def start(self):
+        """
+        Start a thread in the background to process the RDMA tasks and returns.
+        """
+        logger.info("RDMA: starting device processing in the background.")
+        threading.Thread(target=self.process).start()
+
+    def process(self):
+        RDMADeviceHandler.update_dat_conf(dapl_config_paths, self.ipv4_addr)
```

Comment:
On 06/23/2016 01:18 PM, Ahmet Alp Balkan wrote:
>> +    ipv4_addr = None
>> +    mac_adr = None
>> +
>> +    def __init__(self, ipv4_addr, mac_addr):
>> +        self.ipv4_addr = ipv4_addr
>> +        self.mac_addr = mac_addr
>> +
>> +    def start(self):
>> +        """
>> +        Start a thread in the background to process the RDMA tasks and returns.
>> +        """
>> +        logger.info("RDMA: starting device processing in the background.")
>> +        threading.Thread(target=self.process).start()
>> +
>> +    def process(self):
>> +        RDMADeviceHandler.update_dat_conf(dapl_config_paths, self.ipv4_addr)
> 
> @rjschwei I hear from my team that
> 
>> It doesn’t matter which step you do first.
>>
>> If we know this is running in Azure HPC, all those need to be updated to make the whole stack work.
> 
> I don't know what this file does to be honest. If you are sure your suggestion is a better way, we can implement it.
> 

The config files that are being set up configure the OFED driver for the
IB interface.